crates/rpc/rpc/src/eth/filter.rs

```diff
@@ -105,10 +185,59 @@ where
         let mut filters = self.active_filters.inner.lock().await;
         filters.insert(
             id.clone(),
-            ActiveFilter { last_poll_block_number, last_poll_timestamp: Instant::now(), kind },
+            ActiveFilter {
+                block: last_poll_block_number,
+                last_poll_timestamp: Instant::now(),
+                kind,
+            },
         );
         Ok(id)
     }
+
+    /// Returns all logs in the given range that match the filter
+    ///
+    /// Returns an error if:
+    ///  - underlying database error
+    ///  - amount of matches exceeds configured limit
+    #[allow(dead_code)]
+    fn filter_logs(
```

Comment:
we should think about how to make this really fast for large block ranges. wonder if we could return an iterator and stream the results to the rpc client? vs preloading everything server-side?

Reply:
ideally there's a caching layer for blocks